src/openrct2/peep/Guest.cpp

```diff
@@ -1549,7 +1549,7 @@ bool Guest::DecideAndBuyItem(Ride* ride, int32_t shopItem, money32 price)
 loc_69B119:
     if (!hasVoucher)
     {
-        if (price != 0)
+        if ((price != 0) && !(gParkFlags & PARK_FLAGS_NO_MONEY))
```

Comment:
This is a nitpick, but there is no need to put parentheses around `price != 0`.
Same applies to line 2644.